### ISSUE TITLE
Test: Write test successful login user gets redirected to Homepage with username on main navbar

### ISSUE THREAD

**Issue by mtreacy002** [anitab-org/bridge-in-tech-web#33]
## Description
As a developer,
I need test if redirection post successful login works
so that I can make sure user who has successfully logged-in gets redirected to the homepage with the main navbar showing username of login user.

### Update [Required]
- [ ] Write tests using React Testing Library in the Login.test.js file
- [ ] test if user gets redirected to homepage with main navbar showing their username after successful login

### Enhancement to Update [Optional]
NA

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 day

**Comment by Shashwati-tidke** [comment]
@meenakshi-dhanani  Can I work on this issue?

**Comment by Aaishpra** [comment]
Hey @mtreacy002 if no one is working on this can u assign this to me

**Comment by aastha985** [comment]
Hi! @mtreacy002 Can I work on this issue?

**Comment by mtreacy002** [comment]
sure @aastha985. Go ahead 👍.

**Comment by aastha985** [comment]
Hi! I had to update about the status in 3 days. Haven't been able to work on this yet. I will do it by the weekend. I hope that's fine.

**Comment by mtreacy002** [comment]
Yep, sure, @aastha985. Thank you for notifying us on your situation. Let me know if you need any help with the task.

**Comment by aastha985** [comment]
> Yep, sure, @aastha985. Thank you for notifying us on your situation. Let me know if you need any help with the task.

Sure! Thanks😊

**Comment by mtreacy002** [comment]
Hi @aastha985 , I'll make this issue available again since you are now working on backend [issue#70](https://github.com/anitab-org/bridge-in-tech-backend/issues/70)

**Comment by aastha985** [comment]
Sure! @mtreacy002

**Comment by rpattath** [comment]
Hi @mtreacy002, I would like to work on this but have no experience. Looking forward to learning something new.

**Comment by mtreacy002** [comment]
Sure, @rpattath , I'll assign you to this. Let me know if you are facing any issue. we can try work it out together 😉

**Comment by somayaj** [comment]
Can this be worked?

**Comment by mtreacy002** [comment]
@rpattath , are you still working on this issue? Can you please give me an update?

**Comment by mtreacy002** [comment]
Update. This issue is now open to anyone that would like to work on it.

**Comment by nlok5923** [comment]
@mtreacy002 can you please assign this issue to me.

**Comment by Aaishpra** [comment]
@nlok5923 you are already assigned to #131 and according to the guidelines you can be assigned to only one issue at a time in a Repo. So, wait patiently until your PR gets reviewed and merged to claim other issues.😄

**Comment by Bucky25** [comment]
@mtreacy002 i would like to give it a go .

**Comment by mtreacy002** [comment]
Sure, @Bucky25 . I’ll assign you to this. Thanks for picking it up .

**Comment by Bucky25** [comment]
@mtreacy002  i cannot login in the deployed website is there any problem in that . No conformation mail comes to the mail .

**Comment by mtreacy002** [comment]
@Bucky25, are you trying to use option 2 under the Setup section (using BIT and MS servers on Heroku)? Please note the warning that stated Heroku servers are unavailable due to free service limitations. So, the only option left is to setup locally.

**Comment by Bucky25** [comment]
@mtreacy002 i m trying to login on website  https://bridge-in-tech-web-heroku.herokuapp.com/ but after registering no conformation mail comes .

**Comment by mtreacy002** [comment]
Yeah. That’s the Heroku BIT frontend server, one of the servers that are mentioned as not working atm. So, you need to setup your own local servers and database to work on BIT backend and frontend issues.

**Comment by Bucky25** [comment]
@mtreacy002 do all the environment variables have to set or can be remained empty  .
and in local also it tells to verify email before login

**Comment by Bucky25** [comment]
@mtreacy002 can you tell me how to login in locally set up enviroment

**Comment by Bucky25** [comment]
![Screenshot from 2020-12-21 00-52-02](https://user-images.githubusercontent.com/65112369/102722402-e84b4f80-4326-11eb-8810-8fdf1fa840f4.png)
i am getting this error while running python -m unittest discover tests while postrgres is running and ms backend is also  running on ubuntu

**Comment by mtreacy002** [comment]
@Bucky25 . I think this has something to do with machine specific (linux or windows) and how postgresql works on those machines. I had similar issue running the python test from the terminal on Windows machine but no issue (all tests passed) when running test using Test explorer on VS Code. I actually requested anyone with more experience/familiarity with linux/windows to give me some tips on how to resolve the issue (opened issue [#133](https://github.com/anitab-org/bridge-in-tech-backend/issues/133) but still waiting for response. 
Can you see if you can run the test just fine under VS Code test explorer?

**Comment by emmanuelhashy** [comment]
Hi, @mtreacy002  I'd like to work on this.

**Comment by mtreacy002** [comment]
@Bucky25 , where you able to solve the issue? Are you still working on this? If not, can @emmanuelhashy work on it?

**Comment by jalajcodes** [comment]
@mtreacy002  I'd like to work on it.

**Comment by mtreacy002** [comment]
@jalajcodes , I'll remove @Bucky25 assignment to this as there's no reply. But @emmanuelhashy also the next one on the cue. Have you figured out the bug in issue #150 ? Can you open an issue on that? I'll assign you to it.

**Comment by mtreacy002** [comment]
@emmanuelhashy , sorry for the delay. Are you still interested on this issue?

**Comment by jalajcodes** [comment]
@mtreacy002 
No problem. I opened the issue here... https://github.com/anitab-org/bridge-in-tech-web/issues/158
Kindly assign me. Thanks!

**Comment by jalajcodes** [comment]
> @emmanuelhashy , sorry for the delay. Are you still interested on this issue?

@mtreacy002 Maybe I can take it now as Emmanuel doesn't seem to be active?

**Comment by emmanuelhashy** [comment]
@jalaj I am still interested

**Comment by mtreacy002** [comment]
ok, @emmanuelhashy . I'll assign you to this.

**Comment by mtreacy002** [comment]
@emmanuelhashy , any update on this?

**Comment by mtreacy002** [comment]
@jalajcodes , since there's no update from @emmanuelhashy, I'll make this issue available again. Let me know if you're still interested (that's if you're not currently working on other issues 😉).

**Comment by emmanuelhashy** [comment]
No, I am not.

**Comment by mtreacy002** [comment]
@emmanuelhashy, can you give us an update on this then?

**Comment by rkraeher** [comment]
@mtreacy002 Since this is available again, is this issue something doable for a first time contributor? There are currently no unassigned "first timer" issues available.

**Comment by mtreacy002** [comment]
@rkraeher, this issue might be a bit complicated for a first-timer. But if you have some experience in the tech stack and confident you can work on it, I can assign you to this. For their first contribution to the project, a contributor doesn't have to limit themselves to a `First-timers only` issue. That label is there to make sure that the issue is not too challenging for new contributors who might not have the background on the tech stack.

So, let me know if you do want to work on this 😉

**Comment by rkraeher** [comment]
ok thank you. I have experience with React and jest. I looked at the Login.test file and briefly at the React Testing Library docs and I believe I can do it but I was wondering if I can get some clarification on the user story?

Right now there is one test for a successful login testing that an error message doesn't show up (lines 29-46 in Login.test). Am I on the right track to begin something like:
```
it('redirects the user after successful login,' async()=> {
// successful post request
render( <MemberPortfolio /> 
//Rest of the test
)}
```

**Comment by mtreacy002** [comment]
You're on the right target, @rkraeher . Would you like me to assign you to this?

**Comment by rkraeher** [comment]
great! yes can you please assign it to me? thanks!

**Comment by rkraeher** [comment]
If I'm having some trouble setting up the environment should I ask you here or get some help in Zulip?

**Comment by mtreacy002** [comment]
Done, I've assigned you on this @rkraeher . Yep, if you are facing trouble setting up, please reach out to us. Zulip will be a better option for trouble related to setting up, since you'll reach a wider audience compared to here on GitHub. Others who might have experienced the same issue can suggest a solution based on their experience. The post will also be helpful to other contributors that also trying to set up their project. So, I always encourage people to ask questions on zulip for issues related to setting up 😉.

**Comment by mtreacy002** [comment]
@rkraeher , any update on this issue?

**Comment by rkraeher** [comment]
Sorry for the delayed response. I wrote some of this on Zulip but with your help I managed to get the app running in the browser but there is now another missing module error that I need to figure out so I'm still working on getting my environment set up.

**Comment by mtreacy002** [comment]
may I know what's the other missing module, @rkraeher ?

**Comment by rkraeher** [comment]
missing module: psycopg2._psycopg

**Comment by mtreacy002** [comment]
@rkraeher perhaps try installing psycopg2 separately (after the initial dependencies install through requirements.txt) as suggested by @vj-codes [here](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech/topic/Dev.20Env.20Setup/near/211793674) (point 2). Let me know if it works.

**Comment by rkraeher** [comment]
@mtreacy002  Ok I did as you suggested, installed psycopg2 separately, and it is now working. I will now proceed to link up BIT with MS so I have both servers running.

**Comment by mtreacy002** [comment]
@rkraeher , any updates on this issue?

**Comment by rkraeher** [comment]
@mtreacy002 I now have successfully set up both servers and have the frontend site up locally. Starting today I can begin working on writing the test.

**Comment by mtreacy002** [comment]
@rkraeher , any updates on this issue?

**Comment by mtreacy002** [comment]
Putting this on hold following discussion with the contributor on [zulip](https://anitab-org.zulipchat.com/#narrow/stream/237630-bridge-in-tech/topic/help/near/236730451)

**Comment by rkraeher** [comment]
hey there @mtreacy002. i'm feeling better and was interested in getting involved again on the project during hacktober. Can I get back to work on this issue? Or if there are other areas where contribution would be more helpful im flexible and open to suggestion :)

**Comment by vj-codes** [comment]
@rkraeher welcome back 🎉
Removing the issue from 'hold' and you can work on it, have a great start :)

**Comment by rkraeher** [comment]
hi @vj-codes and @mtreacy002 I finally managed to open the PR for this. sorry for the delay. Tagging you both here since the zulip stream seems a bit stale.
